test/test_disjoint_set.jl

```diff
@@ -1,52 +1,70 @@
 @testset "DisjointSet" begin
 
     @testset "IntDisjointSets" begin
-        s = IntDisjointSets(10)
-
-        @testset "basic tests" begin
-            @test length(s) == 10
-            @test eltype(s) == Int
-            @test eltype(typeof(s)) == Int
-            @test num_groups(s) == 10
-
-            for i = 1:10
-                @test find_root(s, i) == i
+        for T in [Int, UInt8, Int8, UInt16, Int16, UInt32, Int32, UInt64]
+            @testset "eltype = $(T)" begin
+                s = IntDisjointSets(T(10))
+                s2 = IntDisjointSets{T}(10)
+
+
+                @testset "basic tests" begin
+                    @test length(s) == 10
+                    @test length(s2) == 10
+                    @test eltype(s) == T
+                    @test eltype(s2) == T
+                    @test eltype(typeof(s)) == T
+                    @test eltype(typeof(s2)) == T
+                    @test num_groups(s) == T(10)
+                    @test num_groups(s2) == T(10)
+
+                    for i = 1:10
```

Comment:
```suggestion
                    for i in 1:10
```

Reply:
in this codebase, the standard way is to use `in` instead of `=` for looping over a range. Please correct this too.

Reply:
OK - but I didn't change this from the original:

https://github.com/JuliaCollections/DataStructures.jl/blob/a4246e7584f3f3d3e94058e686fdf748e5151479/test/test_disjoint_set.jl#L12

Reply:
cool .. leave that .. i can take of it in a separate PR

Reply:
Also, since we're changing the signature of `find_root`, perhaps we should

```
@deprecate find_root find_root!
```
?

Reply:
Sure! I see you have just committed that change.

Reply:
yup. I think everything's ready to go.

Reply:
cool. let me take a quick glance.